com.unity.shadergraph/Editor/Data/Graphs/SamplerStateShaderProperty.cs

```diff
@@ -16,24 +16,24 @@ public SamplerStateShaderProperty()
 
         public override PropertyType propertyType => PropertyType.SamplerState;
 
+        // Sampler States cannot be exposed on a Material
         internal override bool isExposable => false;
-        internal override bool isRenamable => false;
 
-        public override TextureSamplerState value
-        {
-            get => base.value;
-            set
-            {
-                overrideReferenceName = $"SamplerState_{value.filter}_{value.wrap}";
-                base.value = value;
-            }
-        }
+        // subgraph Sampler States can be renamed
+        // just the actual properties they create will always have fixed names
+        internal override bool isRenamable => true;
+
+        internal override bool isReferenceRenamable => false;
```

Comment:
however, the reference name is not renamable (and we override the reference name display to show the propertyName that is actually used)